pkgs/development/compilers/zulu/common.nix

```
@@ -141,6 +141,18 @@ let
         fi
       '';
 
+    doInstallCheck = true;
+    installCheckPhase = ''
+      echo "Checking 'java -version' output contains ${dist.jdkVersion}, 'OpenJDK', and 'Zulu'"
+      output=`$out/bin/java -version 2>&1`
+      # Our Zulu 23 package is a special case (and should be deleted as it is End-of-Life)
+      expected_version=$(if [[ "${dist.jdkVersion}" == 23* ]]; then echo 23; else echo "${dist.jdkVersion}"; fi)
+      echo "Output is: $output"
+      echo "$output" | grep -q "$expected_version"
+      echo "$output" | grep -q "OpenJDK"
+      echo "$output" | grep -q "Zulu"
+    '';
+
```

Nit: you don't need to hand-roll your own bash script here. See https://github.com/NixOS/nixpkgs/blob/master/doc/build-helpers/testers.chapter.md:

```suggestion
  passthru.tests.version = testers.testVersion {
    package = jdk;
    command = "java -version";
    version = ''openjdk version "${if lib.versions.major version == "23" then "23" else version}"'';
  };
```

This does have some subtle differences though:
- It won't look for `OpenJDK` or `Zulu`
- It doesn't get run unless done so explicitly with `nix-build -A zulu.passthru.tests`

---

Oh, that's nice!

Maybe these tests should be `passthru` tests anyway?

Maybe `testVersion` could be extended to also take an optional list of additional keywords (or patterns) that must be found?

---

> it's not that thorough of a functionality test anyways.

Fair enough. We don't _need_ it. I thought it was nice to check that it is actually the distribution of OpenJDK it claims to be. But definitely not necessary.

---

@dtomvan I merged your commit, but it was failing checks, so I added another commit that seems to be working. Apparently the issue with `zulu23` wasn't limited to `zulu23` but can happen on any "nn.0.0" version because the ".0.0" part is not displayed in the version output.

It took me a couple of tries to get this right, but I think the current force-push should pass the tests.